services/lambda-label-successful-prs/CIBot.py

```diff
@@ -0,0 +1,199 @@
+        self.jenkins_username = jenkins_username
+        self.jenkins_password = jenkins_password
+        if apply_secret:
+            self._get_secret()
+        self.all_jobs = None
+        self.jenkins_url = jenkins_url
+        self.prs_labeled = 0
+        self.prs_already_labeled = 0
+        self.prs_status_fail = 0
+        self.prs_unknown_job = 0
+        self.prs_no_build = 0
+
+    def _get_secret(self):
+        """
+        This method is to get secret value from Secrets Manager
+        """
+        secret = json.loads(secret_manager.get_secret())
+        self.github_personal_access_token = secret["github_personal_access_token"]
+        self.jenkins_username = secret["jenkins_username"]
+        self.jenkins_password = secret["jenkins_password"]
+
+    def _get_github_object(self):
+        """
+        This method returns github object initialized with Github personal access token
+        """
+        github_obj = Github(self.github_personal_access_token)
+        return github_obj
+
+    def _find_all_open_prs(self, github_obj):
+        """
+        This method identifies all the open PRs in the given repository
+        It returns a Paginated list of Pull Request objects sorted in
+        descending order of created date i.e. latest PR first
+        :param github_obj
+        """
+        repo = github_obj.get_repo(self.repo)
+        pulls = repo.get_pulls(state='open', sort='created', direction='desc')
+        return pulls
+
+    def _get_jenkins_obj(self):
+        """
+        This method returns an object of Jenkins instantiated using username, password
+        """
+        return Jenkins(self.jenkins_url, username=self.jenkins_username, password=self.jenkins_password)
+
+    def _get_checks(self):
+        """
+        This method finds out all the checks that are currently supported as part of CI
+        """
+        # for now hardcoding list of checks
+        # ideally use Jenkins API to query list of checks and parse it (bit complicated)
+        all_checks = ['clang', 'edge', 'centos-cpu', 'centos-gpu', 'windows-cpu', 'windows-gpu', 'miscellaneous', 'unix-cpu', 'unix-gpu', 'website', 'sanity']
+        return all_checks
+
+    def _is_successful(self, pr):
+        """
+        This method checks if all the pipelines corresponding to a latest build of the PR are successful
+        :param pr PullRequest object
+        """
+        # get jenkins object
+        jenkins_obj = self._get_jenkins_obj()
+
+        # get PR number
+        pr_number = pr.number
+
+        # get a list of checks
+        # Github says "status checks"
+        # Jenkins refers to it as a job name
+        checks = self._get_checks()
+
+        # iterate over the checks
+        for check in checks:
+            # Jenkins Job name
+            name = "mxnet-validation/" + check + "/PR-" + str(pr_number)
+
+            try:
+                # get Jenkins job specific object
+                job = jenkins_obj[name]
```

Comment:
Why check Jenkins at all? All the information is already available through the GitHub checks API. No need to query Jenkins